ext/hal/microchip/mec/mec1501/component/espi_vw.h

```diff
@@ -420,7 +420,7 @@ inline void
 mec_espi_msvw_irq_sel_set(ESPI_MSVW_REG * p, enum espi_msvw_src src,
 			  enum espi_msvw_irq_sel isel)
 {
-	volatile uint8_t *psrc = (volatile uint8_t *)p->SRC_IRQ_SEL;
+	volatile uint8_t *psrc = (volatile uint8_t *)&p->SRC_IRQ_SEL;
```

Comment:
Simplify these 2 lines to 1
p->SRC_IRQ_SEL = (isel << src);

Reply:
The select fields are on byte boundaries and enum espi_msvw_src is one of [0, 1, 2, 3].  Instead:
p->SRC_IRQ_SEL = (isel << (src << 3))
which produces shifts to each byte.

Reply:
Also using a 32-bit write would overwrite the other 3 VWire's IRQ select fields in this 32-bit word. This is one reason I used byte access. The fields are located on byte boundaries so one can use byte access to eliminate read-modify-write. I will be adding comments to the inline functions using this technique.